src/main/java/racing/game/RacingGame.java

```diff
@@ -3,28 +3,48 @@
 import racing.car.CarFactory;
 import racing.car.Cars;
 import racing.view.InputView;
-
+import racing.view.ResultHandler;
+import racing.view.ResultView;
 
 public class RacingGame {
 
+    private final ResultHandler resultHandler;
+    private final Winners winners;
+    private final RacingRandomCalculator racingRandomCalculator;
+
+    RacingGame(ResultHandler resultHandler, Winners winners, RacingRandomCalculator racingRandomCalculator) {
+        this.resultHandler = resultHandler;
+        this.winners = winners;
+        this.racingRandomCalculator = racingRandomCalculator;
+    }
+
     public static void main(String[] args) {
-        InputView input = new InputView();
-        int carNumber = input.carInput();
-        int count = input.cntInput();
+        InputView input = new InputView(); //입력
+        String carName = input.carInput();
+        int cycle = input.cntInput();
 
-        racingCycle(CarFactory.makeCar(carNumber), count);
+        Cars cars = new Cars(CarFactory.generateCars(carName));
+
+        RacingRandomCalculator racingRandomCalculator = new RacingRandomCalculator();
+        Winners winners = new Winners();
+        ResultHandler resultHandler = new ResultView();
+        RacingGame racingGame = new RacingGame(resultHandler, winners, racingRandomCalculator);
+
+        racingGame.racingCycle(cars, cycle); //레이싱
+
+        resultHandler.winnerPrint(winners.getWinners(cars));
     }
 
     /**
      * 주어진 사이클 동안 레이싱하는 기능
      */
-    private static void racingCycle(Cars cars, int count) {
-        RacingCalculator calculator = new RacingCalculator();
-        while (count > 0) {
-            count--;
-            cars.carGoAndStop(calculator);
+    private void racingCycle(Cars cars, int cycle) {
+        while (cycle > 0) {
+            cycle--;
+            cars.carGoAndStop(racingRandomCalculator);
+            resultHandler.racingPrint(cars); //피드백 반영
         }
+        resultHandler.endCycle();
     }
```

Comment:
https://github.com/next-step/java-racingcar/pull/5759/files#r1787948677

지금 구조에선 인터페이스까진 필요없을것 같아요!

```suggestion
    private void racingCycle(Cars cars, int cycle) {
        ResultView resultView = new ResultView();

        while (cycle > 0) {
            cycle--;
            cars.carGoAndStop(racingRandomCalculator);
            resultView.racingPrint(cars);
        }
        resultView.endCycle();
    }
```

위처럼 호출해도 문제없을것 같습니다 :)

이전 피드백 드린부분은 Cars 에서 View 를 의존해서 드린 피드백이었어요